tests/Fixer/ControlStructure/SingleExpressionPerLineFixerTest.php

```
@@ -0,0 +1,329 @@
+ */
+final class SingleExpressionPerLineFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param array<string, mixed> $config
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null, array $config = []): void
+    {
+        $this->fixer->configure($config);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<array{0: string, 1?: null|string, 2?: array<string, mixed>}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield ['<?php $a = [1, 2, 3];'];
+
+        yield [
+            '<?php
+                    $a = [
+1,
+                        2
+];',
+            '<?php
+                    $a = [1,
+                        2];',
+        ];
+
+        yield [
+            '<?php
+                    $a = [
+                        1,
+2,
+                        3
+                    ];',
+            '<?php
+                    $a = [
+                        1, 2,
+                        3
+                    ];',
+        ];
+
+        yield ['<?php $a = array(1, 2, 3);'];
+
+        yield [
+            '<?php
+                    $a = array(
+                        1,
+2,
+                        3
+                    );',
+            '<?php
+                    $a = array(
+                        1, 2,
+                        3
+                    );',
+        ];
+
+        yield ['<?php $a = foo(1, 2, 3);'];
+
+        yield [
+            '<?php
+                    $a = foo(1, 2,
+                        3
+                    );',
+        ];
+
+        yield [
+            '<?php
+                    $a = foo(
+1,
+2,
+                        3
+                    );',
+            '<?php
+                    $a = foo(1, 2,
+                        3
+                    );',
+            ['elements' => [SingleExpressionPerLineFixer::ELEMENTS_ARGUMENTS]],
```

I think it's better not to use such constants here and to use hardcoded values instead. You want your tests to fail if the accepted values change in the implementation, which is not possible when the implementation's constants are also used in the test.

Then maybe the constants could be made private in the implementation.